packages/web/src/common/store/cache/sagas.ts

```diff
@@ -105,20 +110,25 @@ export function* retrieve({
   deleteExistingEntry = false,
   onBeforeAddToCache = function* (_metadatas) {},
   onAfterAddToCache = function* (_metadatas) {}
-}: RetrieveArgs) {
+}: RetrieveArgs<T>): Generator<
+  any,
+  { entries: Record<ID | string, T>; uids: Record<ID | string, string> }
+> {
   if (!ids.length) {
     return {
-      entries: [],
-      uids: []
+      entries: {},
+      uids: {}
```

Comment:
was this a bug previously? Or did you just change the return type

Reply:
see the comment I left. It was a bug, we want a id -> uid lookup